src/signtool/IBM_Container.cpp

```diff
@@ -0,0 +1,1767 @@
+        } 
+
+        THROW_EXCEPTION(p_buffer.size() != ECDSA521_SIG_SIZE);
+    } 
+}
+
+
+
+// ContainerHdr C'tor
+ContainerHdr::ContainerHdr()
+   : m_magicNumber( ROM_MAGIC_NUMBER ),
+     m_version( CONTAINER_VERSION ),
+     m_containerSize(0),
+     m_targetHrmor(0),
+     m_stackPointer(0),
+     m_hwPkeyA(),
+     m_hwPkeyB(),
+     m_hwPkeyC()
+{
+}
+
+
+/**
+ * @brief   Get the Container Header message as a byte stream
+ *
+ * @param[out]  packet  -  A reference to a vector of type uint8_t
+ */
+void ContainerHdr::GetHeaderBytes( std::vector<uint8_t>& packet ) const
+{
+    packet.clear();
+
+    // insert the magic nummber field
+    packet.push_back((m_magicNumber >> 24) & 0xFF);
+    packet.push_back((m_magicNumber >> 16) & 0xFF);
+    packet.push_back((m_magicNumber >> 8) & 0xFF);
+    packet.push_back(m_magicNumber & 0xFF);
+
+    // insert the version field
+    packet.push_back((m_version >> 8) & 0xFF);
+    packet.push_back(m_version & 0xFF);
+
+    // insert the container size field
+    packet.push_back((m_containerSize >> 56) & 0xFF);
+    packet.push_back((m_containerSize >> 48) & 0xFF);
+    packet.push_back((m_containerSize >> 40) & 0xFF);
+    packet.push_back((m_containerSize >> 32) & 0xFF);
+    packet.push_back((m_containerSize >> 24) & 0xFF);
+    packet.push_back((m_containerSize >> 16) & 0xFF);
+    packet.push_back((m_containerSize >> 8) & 0xFF);
+    packet.push_back(m_containerSize & 0xFF);
+
+    // insert the target Hrmor field
+    packet.push_back((m_targetHrmor >> 56) & 0xFF);
+    packet.push_back((m_targetHrmor >> 48) & 0xFF);
+    packet.push_back((m_targetHrmor >> 40) & 0xFF);
+    packet.push_back((m_targetHrmor >> 32) & 0xFF);
+    packet.push_back((m_targetHrmor >> 24) & 0xFF);
+    packet.push_back((m_targetHrmor >> 16) & 0xFF);
+    packet.push_back((m_targetHrmor >> 8) & 0xFF);
+    packet.push_back(m_targetHrmor & 0xFF);
+    
+    // insert the stack Pointer field
+    packet.push_back((m_stackPointer >> 56) & 0xFF);
+    packet.push_back((m_stackPointer >> 48) & 0xFF);
+    packet.push_back((m_stackPointer >> 40) & 0xFF);
+    packet.push_back((m_stackPointer >> 32) & 0xFF);
+    packet.push_back((m_stackPointer >> 24) & 0xFF);
+    packet.push_back((m_stackPointer >> 16) & 0xFF);
+    packet.push_back((m_stackPointer >> 8) & 0xFF);
+    packet.push_back(m_stackPointer & 0xFF);
+    
+    // insert the HW Public Key-A field
+    packet.insert( packet.end(),
+                   m_hwPkeyA,
+                   (m_hwPkeyA + sizeof(m_hwPkeyA)) );
+    
+    // insert the HW Public Key-B field
+    packet.insert( packet.end(),
+                   m_hwPkeyB,
+                   (m_hwPkeyB + sizeof(m_hwPkeyB)) );
+    
+    // insert the HW Public Key-C field
+    packet.insert( packet.end(),
+                   m_hwPkeyC,
+                   (m_hwPkeyC + sizeof(m_hwPkeyC)) );
+}
+
+
+/**
+ *  @brief   print the Container Header
+ */
+void ContainerHdr::PrintHeader() const
```

Comment:
Why not do the << magic and then you're not forced to print to cout, but can instead just redirect to any ostream ?